cmd/repo_review.go

```diff
@@ -0,0 +1,154 @@
+           - Key improvement areas`,
+	Args: cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		owner, repo := args[0], args[1]
+		saveOutput, err := cmd.Flags().GetBool("output")
+		if err != nil {
+			color.Red("Error: %v", err)
+			return
+		}
+
+		color.Blue("\n🔍 Fetching all source code files recursively...\n")
+		files, err := github_internal.GetRepositoryFilesRecursive(owner, repo)
+		if err != nil {
+			color.Red("❌ ERROR: Fetching repository files: %s\n", err)
+			return
+		}
+
+		color.Green("📂 Repository contains %d files\n", len(files))
+
+		sourceCode := fetchAllSourceCode(owner, repo, files)
+		if len(sourceCode) == 0 {
+			color.Red("❌ ERROR: No valid source code found for analysis")
+			return
+		}
+
+		color.Blue("\n🤖 Training AI with full source code...\n")
+		summary, useCases, codeReview, securityReview, improvements := analyzeRepositoryWithAI(sourceCode)
+
+		displayRepoAnalysis(repo, summary, useCases, codeReview, securityReview, improvements)
+
+		if saveOutput {
+			saveRepoAnalysis(repo, owner, summary, useCases, codeReview, securityReview, improvements)
+		}
+	},
+}
+
+func fetchAllSourceCode(owner, repo string, files []string) string {
+	var allCode strings.Builder
+
+	for _, file := range files {
+		if strings.HasSuffix(file, ".md") || strings.Contains(file, "LICENSE") {
+			continue
+		}
+
+		color.Cyan("\n📄 Reading file: %s", file)
+
+		content, err := github_internal.GetFileContent(owner, repo, file)
+		if err != nil {
+			color.Red("❌ ERROR fetching file content: %s\n", err)
+			continue
+		}
+
+		allCode.WriteString(fmt.Sprintf("\n// File: %s\n%s\n", file, content))
+	}
+
+	return allCode.String()
+}
+
+func analyzeRepositoryWithAI(sourceCode string) (string, string, string, string, string) {
+	summaryPrompt := "Analyze this entire source codebase and provide a concise summary of what the application does."
+	useCasesPrompt := "Extract the most important use cases from the source code."
+	codeQualityPrompt := "Identify the most critical code quality issues found in the source code. Provide a brief list."
+	securityPrompt := "Identify the most critical security vulnerabilities in the source code. Provide a brief list."
+	improvementPrompt := "Suggest the most important areas to improve in the application."
+
+	summary, _ := openai.AnalyzeCodeWithAI(sourceCode, summaryPrompt)
+	useCases, _ := openai.AnalyzeCodeWithAI(sourceCode, useCasesPrompt)
+	codeReview, _ := openai.AnalyzeCodeWithAI(sourceCode, codeQualityPrompt)
+	securityReview, _ := openai.AnalyzeCodeWithAI(sourceCode, securityPrompt)
+	improvements, _ := openai.AnalyzeCodeWithAI(sourceCode, improvementPrompt)
+
+	return summary, useCases, codeReview, securityReview, improvements
+}
+
+func displayRepoAnalysis(repo, summary, useCases, codeReview, securityReview, improvements string) {
+	color.Magenta("\n📌 Repository Analysis Summary for %s\n", repo)
+	color.Cyan("\n📖 Application Summary:\n")
+	fmt.Println(summary)
```

Comment:
2. In the same file, within the `displayRepoAnalysis` function, we directly print out the analysis result without checking if they are empty. It might be good to verify these variables prior to printing.